sgkit/io/vcf/vcf_reader.py

```diff
@@ -401,6 +401,9 @@ def vcf_to_zarr(
             temp_chunk_length=temp_chunk_length,
             tempdir=tempdir,
```

Comment:
It might be nice to try and avoid this problem in the future via something like

```python
if stuff:
    convert_func = vcf_to_zarr_sequential
else:
    convert_func = functools.partial(
             vcf_to_zarr_parallel, 
             temp_chunk_length=temp_chunk_length,
￼            tempdir=tempdir,
            tempdir_storage_options=tempdir_storage_options)
convert_func(
           input,
           output,
           region=regions,
            chunk_length=chunk_length,
            chunk_width=chunk_width,
            ploidy=ploidy,
            mixed_ploidy=mixed_ploidy,
            truncate_calls=truncate_calls
)
```
(I haven't tried this, so might not work, but you get the idea!)

Reply:
Thanks @jeromekelleher this is much cleaner. Only minor change is `regions` needs to be passed by position because of the singular vs plural argument for each function.

Reply:
Excellent!